Shared/Models/WriteFreelyModel.swift

```diff
@@ -6,28 +6,48 @@ import Network
 // MARK: - WriteFreelyModel
 
 final class WriteFreelyModel: ObservableObject {
+
+    // MARK: - Models
     @Published var account = AccountModel()
     @Published var preferences = PreferencesModel()
     @Published var posts = PostListModel()
     @Published var editor = PostEditorModel()
+
+    // MARK: - Error handling
+    @Published var hasError: Bool = false
+    var currentError: Error? {
+        didSet {
+            #if DEBUG
+            print("⚠️ currentError -> didSet \(currentError?.localizedDescription ?? "nil")")
+            print("  > hasError was: \(self.hasError)")
+            #endif
+            DispatchQueue.main.async {
+                #if DEBUG
+                print("  > self.currentError != nil: \(self.currentError != nil)")
+                #endif
+                self.hasError = self.currentError != nil
+                #if DEBUG
+                print("  > hasError is now: \(self.hasError)")
+                #endif
+            }
+        }
+    }
```

Comment:
Currently we're printing some helpful debug messages to the console when the `currentError` is set; I plan to remove all this when #204 is finalized.